### DISCUSSION TITLE
Can AqualinkD use just Extended Device ID's without the RS Allbutton mode?

### DISCUSSION THREAD

**Discussion by niharmehta** [aqualinkd/AqualinkD#258] [General]
I am still commonly plagued by the SWG bounces because of the issues around Pentair VSP dropping offline (well, the Jandy thinks it is) . When I run AqualinkD without the extended device settings, the SWG is much more stable. However, I no longer have VS control.  
I also have noticed that sometimes if I swap between the OneTouch (0x40) and AqualinkTouch  modes (0x30) , it will be more stable for a while when just a simple restart does not do the same. It makes me suspect that something is being held in memory on the RS.
Anyways, my question is that if the Jandy can be fully controlled with only a AqualinkTouch or OneTouch natively, why not disable the AllButton device mode?  I understand there are some performance tradeoffs possibly, but I wonder if this may be triggering some bug in the RS.

**Comment by sfeakes** [comment]
The all button protocol is the only protocol that's supported on all Jandy RS control panels (except PDA only panels), so to make AqualinkD support as much as possible it uses that protocol as its core.   It wouldn't be easy for AqualinkD to "not use it", and if you did, you'd be left with similar functionality as PDA only panel.   There are a few things I've been playing with on the issue you mention, but I don't think any will be a quick fix.

**Comment by ballle98** [comment]
@niharmehta what type of raspberry pi are you using?  I noticed instability with PI 4 see #259.  You could force it into PDA mode using aqualinkd.conf.  If you do install my branch which has some fixes for SWG.  I do not have VSP so there is probably no support for that in PDA mode yet.

https://github.com/ballle98/AqualinkD/wiki#install

**Comment by niharmehta** [comment]
@ballle98 , I am currently using a BananaPi Zero for the last couple of years.  (I smoked my previous RPi Zero and it was the only similar option I could get at the time).  I am about to move over to a new  Pi Zero 2  and use wired ethernet vs. wifi. 
I will test your branch on my new Pi. Although no VSP support will be an issue long term.

**Comment by niharmehta** [comment]
@sfeakes thanks for the info. Having not had a detailed look at the code, I was not sure if the entire command set for OneTouch or iAqualink was  implemented or just the extended ones beyond AllButton,  so I was hoping that there was a simple way to just disable AllButton and run the newer protocols natively on a supported board.
